File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Dictionary;
import java.util.StringTokenizer;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;


class Stack {
	private int [] intArray;
	private int index;
	StringBuilder sb = new StringBuilder();
	public Stack(int N) {
		intArray = new int [N];
		index = 0;
	}
	void push(int X) {
		intArray[index] = X;
		index++;
	}
	void pop() {
		if(index>0) {
			index--;
			sb.append(intArray[index]+"\n");
		}
		else {
			sb.append(-1+"\n");
		}
		
	}
	void size() {
		sb.append(index+"\n");
	}
	void empty() {
		if(index==0) {
			sb.append(1+"\n");
		}
		else {
			sb.append(0+"\n");
		}
	}
	void top() {
		if(index>0) {
			sb.append(intArray[index-1]+"\n");
		}
		else {
			sb.append(-1+"\n");
		}
	}
}
public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		Stack s = new Stack(N);
		for(int i=0;i<N;i++) {
			String command = br.readLine();
			
			switch(command) {
				case "pop":
					s.pop();
					break;
				case "size":
					s.size();
					break;
				case "empty":
					s.empty();
					break;
				case "top":
					s.top();
					break;
				default:
					String [] push = command.split(" ");
					s.push(Integer.parseInt(push[1]));
					break;
			}
		}
		System.out.print(s.sb);
		br.close();
	}
}

File: 백준/Silver/10828. 스택/스택.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main(args: Array<String>) = with(BufferedReader(InputStreamReader(System.`in`))) {
    // store input data
    val N = readLine().toInt()
    val inputs = (1..N).map { readLine() }
    // create stack and execute
    val stack = Stack()
    stack.execute(inputs)
}

class Stack() {
    private val elements = ArrayList<Int>()
    private val log = StringBuilder()

    fun execute(commands: List<String>) {
        // execute commands
        commands.forEach { command ->
            when {
                command.contains("push") -> push(command.split(" ")[1].toInt())
                command.equals("pop", ignoreCase = true) -> pop()
                command.equals("size", ignoreCase = true) -> size()
                command.equals("empty", ignoreCase = true) -> empty()
                command.equals("top", ignoreCase = true) -> top()
            }
        }
        // print log
        printLog()
    }

    // add x to stack
    private fun push(x: Int) {
        elements.add(x)
    }

    // pop last element and print it if there is no element print -1
    private fun pop() {
        log.append(if(elements.size == 0) -1 else elements.last()).append("\n")
        if (elements.size != 0) { elements.removeLast() }
    }

    // print stack size
    private fun size() {
        log.append(elements.size).append("\n")
    }

    // if stack is empty print 1 else 0
    private fun empty(): Boolean {
        log.append(if(elements.size == 0) 1 else 0).append("\n")
        return elements.size == 0
    }

    // print last element if there is no element print -1
    private fun top() {
        log.append(if(elements.size == 0) -1 else elements.last()).append("\n")
    }

    // print all log
    private fun printLog() {
        println(log)
    }
}